Address consistency

jaw6 opened the pull request:
Based on comments from @brntbeer in https://github.com/github/training-utils/pull/4, I'm attempting to make two new-ish Ruby scripts more consistent with respect to one another. Mainly, this changes `add-collaborators` so that it accepts and parses an issue URL (eg, instead of having to separate into `-r repo -i issue`).

Fwiw, I didn't add the `-r -i` flags to `create-assigned-issue` since the "template issue" is core to the concept there. Maybe eventually (if folks are :+1:), we could just move to issue URLs entirely (eg, we can get rid of the `-r -i` altogether). Let me know if folks disagree with that, it's not hard to build that in.

Commit:
File extension arguably unnecessary

Commit:
File extension arguably unnecessary

Commit:
Fixup create

Commit:
Make add-collaborators executable

Commit:
Cleanup and move common code to common.rb

Commit:
Add issue-url parsing to add-collaborators

Other minor changes to make this more consistent with create-assigned-user

jaw6 commented:
/cc @patrickmckenna (I believe this was originally your script? I don't want to step on anyone's toes here!)

patrickmckenna commented:
@jaw6 No toes stepped on, I appreciate seeing what a Rubyist would change! Incidentally, @loranallensmith modified my original script, but doubt he'd be upset about these edits.

Commit:
But actually work correctly

Turns out this only works when run from the local directory -- once I installed this into ~/bin (I do this in my demo user account) it stopped working altogether.

jaw6 commented:
Looks like there aren't any major outstanding issues here, I'm going to merge this now.
